Userland/Libraries/LibWebView/AccessibilityTreeModel.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2022, Jonah Shafran <jonahshafran@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWebView/AccessibilityTreeModel.h>
+
+namespace WebView {
+
+AccessibilityTreeModel::AccessibilityTreeModel(JsonObject accessibility_tree, GUI::TreeView* tree_view)
+    : m_tree_view(tree_view)
+    , m_accessibility_tree(move(accessibility_tree))
+{
+    map_accessibility_nodes_to_parent(nullptr, &m_accessibility_tree);
+}
+
+AccessibilityTreeModel::~AccessibilityTreeModel() = default;
+
+GUI::ModelIndex AccessibilityTreeModel::index(int row, int column, const GUI::ModelIndex& parent) const
+{
+    if (!parent.is_valid()) {
+        return create_index(row, column, &m_accessibility_tree);
+    }
+
+    auto const& parent_node = *static_cast<JsonObject const*>(parent.internal_data());
+    auto const* children = get_children(parent_node);
+    if (!children)
+        return create_index(row, column, &m_accessibility_tree);
+
+    auto const& child_node = children->at(row).as_object();
+    return create_index(row, column, &child_node);
+}
+
+GUI::ModelIndex AccessibilityTreeModel::parent_index(const GUI::ModelIndex& index) const
+{
+    if (!index.is_valid())
+        return {};
+
+    auto const& node = *static_cast<JsonObject const*>(index.internal_data());
+
+    auto const* parent_node = get_parent(node);
+    if (!parent_node)
+        return {};
+
+    // If the parent should not be displayed, the parent is actually the grandparent
+    if (parent_node->get("type"sv).as_string() == "excluded")
+        parent_node = get_parent(*parent_node);
```

Comment:
Should this be a loop? What happens if the grandparent is also marked as "excluded"? (Assuming that's even possible)

Reply:
Removed, this was from an earlier iteration where we were doing this filtering in the AccessibilityTreeModel, it proved simpler to do it when generating the tree in Node::build_accessibility_tree.